includes/classes/db/mysql/query_factory.php

```diff
@@ -964,10 +964,31 @@ function __construct($field)
         $this->type = $matches[0];
         $this->max_length = preg_replace('/[a-z\(\)]/', '', $type);
         if (empty($this->max_length)) {
-            if (strtoupper($type) === 'DATE') $this->max_length = 10;
-            if (strtoupper($type) === 'DATETIME') $this->max_length = 19; // ignores fractional which would be 26
-            if (strtoupper($type) === 'TIMESTAMP') $this->max_length = 19; // ignores fractional which would be 26
-            if (strtoupper($type) === 'TINYTEXT') $this->max_length = 255; 
+           switch (strtoupper($type)) {
+              case 'DATE':
+                  $this->max_length = 10;
+                  break;
+              case 'DATETIME':
+              case 'TIMESTAMP':
+                  $this->max_length = 19; // ignores fractional which would be 26
+                  break;
+              case 'TINYTEXT':
+                  $this->max_length = 255;
+                  break;
+              case 'INT':
+                  $this->max_length = 11;
+                  break;
+              case 'TINYINT':
+                  $this->max_length = 4;
+                  break;
+              case 'SMALLINT':
+                  $this->max_length = 4;
+                  break;
+              default:
+                  $this->max_length = 8; // should not get here unless db changed 
```

Comment:
Where did this come from and why 8?  If you're going to provide a default value, a notification is in order to allow site-specific updates to deal with their changes.

Reply:
If you don't have a value, you'll get a fatal error.   I can take out the default branch if you want but this is really anti-bugging - if 8 is too short, they'll file a bug report and we'll know we have to add a new case.

Reply:
I'd be happy to change it to 64 or whatever magic number you think is ideal; I just don't want someone's site to suddenly crash.

Reply:
Please add a notification after setting that default value, you'll need to add `extends base` to the `queryFactoryMeta` class statement:
```
$this->notify('NOTIFY_QUERY_FACTORY_META_DEFAULT', ['field' => $field, 'type' => $type], $this->max_length);
```

That will enable site-specific additions to (a) make their required modifications and/or (b) log a PHP notice to identify the missing type.